packages/react/menu/src/Menu.tsx

```diff
@@ -956,6 +1012,30 @@ function getNextMatch(values: string[], search: string, currentMatch?: string) {
   return nextMatch !== currentMatch ? nextMatch : undefined;
 }
 
+type Point = { x: number; y: number };
+type Triangle = [Point, Point, Point];
+
+function getTriangleArea(a: Point, b: Point, c: Point) {
+  return Math.abs((a.x * (b.y - c.y) + b.x * (c.y - a.y) + c.x * (a.y - b.y)) / 2);
+}
+
+function isPointInTriangle(point: Point, a: Point, b: Point, c: Point) {
+  const A = getTriangleArea(a, b, c);
+  const A1 = getTriangleArea(point, b, c);
+  const A2 = getTriangleArea(a, point, c);
+  const A3 = getTriangleArea(a, b, point);
+  return A === A1 + A2 + A3;
+}
```

Comment:
🤯  obviously just going to have to trust you on this one, way over my head haha

Reply:
Don't worry we copied it from reakit 😀

Reply:
The overall formula makes sense though when you think about it, adding the area of each triangle formed using the point would be exactly the area of the triangle only if it's inside, otherwise it would be bigger.